amazon-kinesis-connector-flink/src/main/java/software/amazon/kinesis/connectors/flink/FlinkKinesisConsumer.java

```diff
@@ -462,6 +465,25 @@ public void snapshotState(FunctionSnapshotContext context) throws Exception {
 		}
 	}
 
+	/**
+	 * Creates state serializer for kinesis shard sequence number.
+	 * Using of the explicit state serializer with KryoSerializer is needed because otherwise
+	 * users cannot use 'disableGenericTypes' properties with KinesisConsumer, see FLINK-24943 for details
+	 *
+	 * @return state serializer
+	 */
+	@VisibleForTesting
+	static TupleSerializer<Tuple2<StreamShardMetadata, SequenceNumber>> createStateSerializer(ExecutionConfig executionConfig) {
```

Comment:
Why make this static?

Reply:
To simplify calling from test. I can move to a dedicated class, that should also solve the problem of @VisibleForTesting annotation usage.